### PR TITLE
added gainControl - allow end user to update master gain

### PR DESCRIPTION
**Pull request by nspilman**

### CODE REVIEW

File: src/components/gainControl.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    const setGainAndState = (newValue: number) => {
+        setGain(audioContextWrapper, newValue)
+        setState({ ...state, gain: newValue })
+    }
+
+    const { audioContextWrapper, gain } = state;
+
+    const decrementGain = () => {
+        const newValue = (gain - 1 < minOctave) ? minOctave : gain - 1;
```
Comment (soiferj) [inline src/components/gainControl.tsx]: minGain?

File: src/components/gainControl.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        setGainAndState(newValue)
+    }
+
+    const incrementGain = () => {
+        const newValue = (gain + 1 > maxOctave) ? maxOctave : gain + 1;
```
Comment (soiferj) [inline src/components/gainControl.tsx]: maxGain?

Comment (nspilman) [reply src/components/gainControl.tsx]: fixed!

File: src/components/gainControl.tsx (hunk 1 above)
Comment (nspilman) [reply src/components/gainControl.tsx]: fixed!

**Pull request merged**